app/controllers/assignments_controller.rb

```diff
@@ -92,7 +95,35 @@ def create
       @assignment.available = @assignment.due_date
     end
 
-    if @assignment.save
+    # Options are:
+    # None -- create a (unique, ignored) teamset
+    # New -- create a new teamset with initially no teams
+    # Use -- share an existing teamset with another assignment
+    # Copy -- duplicate an existing teamset
+    
+    @assignment.team_subs = (params["teamset"] != "none")
```

Comment:
It means the team set is potentially wrong before you save, which is true for longer in your version. Buf if you're not transactional, it means your *persisting* inconsistent state, which is a much more serious problem.